### DISCUSSION TITLE
IgxDateTimeEditor: valueChange type

### DISCUSSION THREAD

**Discussion by Timmeeeey** [IgniteUI/igniteui-angular#12126] [General]
The IgxDateTimeEditor `valueChange` Event has the type `Date | string` but it only emits `Date` and never `string`.
Is this a bug or intentional for some reason?

**Comment by jackofdiamond5** [comment]
Hi, The `valueChange` event is primarily intended to make [two-way binding](https://angular.io/guide/two-way-binding#how-two-way-binding-works) with the `value` property possible. It does emit a date object only as this is what the directive is internally using. Though, it may be a bit confusing for it to be a union between `Date` and `string` as it never really emits `string` values.  I've also logged an [issue](https://github.com/IgniteUI/igniteui-angular/issues/12132) about a little problem that I discovered while poking around which is related to two-way binding with the `IgxDateTimeEditorDirective` in the context of an `IgxInputGroup`.
